pychromecast/dial.py

```diff
@@ -49,24 +57,30 @@ def get_host_from_service(service, zconf):
     return _get_host_from_zc_service_info(service_info) + (service_info,)
 
 
-def _get_host_from_zc_service_info(service_info: zeroconf.ServiceInfo):
+def _get_host_from_zc_service_info(
+    service_info: zeroconf.ServiceInfo | None,
+) -> tuple[str | None, int | None]:
     """Get hostname or IP + port from zeroconf service_info."""
     host = None
     port = None
-    if (
-        service_info
-        and service_info.port
-        and (service_info.server or len(service_info.addresses) > 0)
-    ):
+    if service_info and service_info.port:
```

Comment:
There's no risk that the port is 0, right?

Reply:
No, port 0 is not a valid port.